gtsam/discrete/DecisionTreeFactor.cpp

```diff
@@ -167,7 +167,10 @@ namespace gtsam {
   DiscreteKeys DecisionTreeFactor::discreteKeys() const {
     DiscreteKeys result;
     for (auto&& key : keys()) {
-      result.emplace_back(key, cardinality(key));
+      DiscreteKey dkey(key, cardinality(key));
```

Comment:
Why? This should never be triggered, and we should get to root cause of the cases that *would* trigger this.

Reply:
The lack of this causes a test to fail since we get M1 twice. There is double counting because both keys and discreteKeys in the DCFactorGraph have M1.